examples/vision/retinanet.py

```diff
@@ -0,0 +1,700 @@
+
+    return _add_l2_regularization
+
+
+"""
+## Setting up training parameters
+"""
+
+model_dir = "retinanet/"
+label_encoder = LabelEncoder()
+
+num_classes = 80
+batch_size = 2
+
+learning_rates = [2.5e-06, 0.000625, 0.00125, 0.0025, 0.00025, 2.5e-05]
+learning_rate_boundaries = [125, 250, 500] + [240000, 360000]
+learning_rate_fn = tf.optimizers.schedules.PiecewiseConstantDecay(
+    boundaries=learning_rate_boundaries, values=learning_rates
+)
+
+"""
+## Initializing and compiling model
+"""
+
+loss_fn = RetinaNetLoss(num_classes)
+model = RetinaNet(num_classes)
+
+
+#  Add L2 regularization to all Conv2d weights
+for weight in model.trainable_weights:
+    if re.search(r"^(conv).*(kernel:0)$", weight.name) is not None:
+        model.add_loss(add_l2_regularization(weight))
+
+optimizer = tf.optimizers.SGD(learning_rate=learning_rate_fn, momentum=0.9)
+model.compile(loss=loss_fn, optimizer=optimizer)
+
+"""
+## Setting up callbacks
+"""
+
+callbacks_list = [
+    tf.keras.callbacks.ModelCheckpoint(
+        filepath=os.path.join(model_dir, "weights" + "_epoch_{epoch}"),
+        monitor="loss",
+        save_best_only=False,
+        save_weights_only=True,
+        verbose=1,
+    )
+]
+
+"""
+## Load the COCO2017 dataset using TensorFlow Datasets
+"""
+
+#  set `data_dir=None` to load the complete dataset
+
+(train_dataset, val_dataset), dataset_info = tfds.load(
+    "coco/2017", split=["train", "validation"], with_info=True, data_dir="data"
+)
+
+"""
+## Setting up tf.data pipeline
+"""
+
+autotune = tf.data.experimental.AUTOTUNE
+train_dataset = train_dataset.map(preprocess_data, num_parallel_calls=autotune)
+train_dataset = train_dataset.shuffle(128)
+train_dataset = train_dataset.padded_batch(
+    batch_size=batch_size, padding_values=(0.0, 1e-8, -1), drop_remainder=True
+)
+train_dataset = train_dataset.map(
+    label_encoder.encode_batch, num_parallel_calls=autotune
+)
+train_dataset = train_dataset.apply(tf.data.experimental.ignore_errors())
+train_dataset = train_dataset.repeat()
+train_dataset = train_dataset.prefetch(autotune)
+
+val_dataset = val_dataset.map(preprocess_data, num_parallel_calls=autotune)
+val_dataset = val_dataset.padded_batch(
+    batch_size=1, padding_values=(0.0, 1e-8, -1), drop_remainder=True
+)
+val_dataset = val_dataset.map(label_encoder.encode_batch, num_parallel_calls=autotune)
+val_dataset = val_dataset.apply(tf.data.experimental.ignore_errors())
+val_dataset = val_dataset.prefetch(autotune)
+
+"""
+## Training the model
+"""
+
+# Uncomment the following lines, when training on full dataset
+# train_steps_per_epoch = dataset_info.splits["train"].num_examples // batch_size
+# val_steps_per_epoch = \
+#     dataset_info.splits["validation"].num_examples // batch_size
+
+# train_steps = 4 * 100000
+# epochs = train_steps // train_steps_per_epoch
+# validation_freq = 5,
+
+epochs = 1
+train_steps_per_epoch = 100
+val_steps_per_epoch = 100
+validation_freq = 1
+
+model.fit(
```

Comment:
IIUC we usually needs to include a snippet of training summaries?

Reply:
I thought about plotting the training loss graph, but this script by default runs only for 100 steps and i don't think there would be anything interesting in looking at training loss for the first 100 steps. If you had something else in mind please let me know!

Reply:
model fit prints everything to the console and should give you *all* the data? If not you can always do `history=model.fit` and try to save/plot `history`

Reply:
Yes, IIUC the script already prints out what model.fit is sending to console